artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/OpenWireConnection.java

```diff
@@ -213,6 +239,7 @@ public OpenWireConnection(Connection connection,
       this.useKeepAlive = openWireProtocolManager.isUseKeepAlive();
       this.maxInactivityDuration = openWireProtocolManager.getMaxInactivityDuration();
       this.transportConnection.setProtocolConnection(this);
+      this.maxActorSize = openWireProtocolManager.getMaxActorSize();
```

Comment:
I would use an actor which queue is bounded (we have many in JCtools) adding a method boolean tryAct
that return false if the queue has reached maximum capacity.
This would be more performant and easy to manage, given that could be used by other part of code to limit the size of tasks on the actor.

Reply:
the thing is, I may have to go beyond the limit some times.. because the parse on the packet already happened. and the messages were already read from the network. I have to read them all.